src/mpid/ch4/shm/posix/posix_coll_release_gather.h

```diff
@@ -123,15 +126,15 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_bcast_release_gather(void *buffer,
     /* Print chunking information */
 /* *INDENT-OFF* */
     MPL_DBG_MSG_FMT(MPIR_DBG_COLL, VERBOSE, (MPL_DBG_FDEST, "Bcast shmgr pipeline info: segsize=%d\
-                                             count=%d num_chunks=%d chunk_count_floor=%d\
-                                             chunk_count_ceil=%d \n",
-                                             cellsize, (int) (count * type_size), num_chunks,
+                                             count=%ld num_chunks=%ld chunk_count_floor=%ld\
```

Comment:
Because the underlying type of `MPI_Aint` depends on the platform, MPICH provides these format specifiers for printing values: `MPI_AINT_FMT_DEC_SPEC`. We should use it instead of `%ld`.

Reply:
Another option is to use `%ld` then cast the following argument with `(long)`. That may result in truncation on some platforms, but it should be good enough for debug messages.  Alternatively, we can use `%lld` and cast to `(long long)`.

`MPI_AINT_FMT_DEC_SPEC` is not in the standard, right? So it may disappear at some point.

Reply:
Right, it's not standard. The fact that MPICH provides it in mpi.h and has MPI_ prefix is not great, but isn't doing much harm. It might be a good idea to shift the namespace to MPIX if the community doesn't object.

Reply:
I see that `MPI_AINT_FMT_DEC_SPEC` is few other places, hence went ahead and used it for this as well.